app/actions/remote/thread.ts

```diff
@@ -49,7 +49,7 @@ export const fetchAndSwitchToThread = async (serverUrl: string, rootId: string)
         const post = await getPostById(database, rootId);
         if (post) {
             const thread = await getThreadById(database, rootId);
-            if (thread?.unreadReplies || thread?.unreadMentions) {
+            if (thread && (isFromNotification || thread.unreadReplies || thread.unreadMentions)) {
```

Comment:
the `isFromNotification` should be passed down to `switchToThread` but you really don't need this if as we should always mark the thread as read